### PR TITLE
fix(discover): add share button and update api endpoint [SOC-99]

### PR DESCRIPTION
**Pull request by anthony-liddle**
## Goal

Add the share button to the recommendations on the Discover page

## To Do:

- [x] Add share button to recommendations
- [x] Ensure clicking the button copies the link to the clipboard
- [x] BONUS: Update the recs API endpoint 

## Implementation Decisions

Added the Share button. Wrapped in an extra div in preparation for the next icon/action to be included in that chunk.

### CODE REVIEW

File: components/timeline/TimelineDiscover.vue [hunk 1]
```diff
@@ -18,6 +18,13 @@ function shorten(str: string, maxLen: number): string {
 function updateUTM(url: string): string {
   return url.replace('pocket-newtab', 'mozilla')
 }
+
+const clipboard = useClipboard()
```
Comment (jpezninjo) [inline components/timeline/TimelineDiscover.vue:22]: not having to import this still throws me off so much coming from `Web` and `web-client` land :')

File: components/timeline/TimelineDiscover.vue [hunk 2]
```diff
@@ -48,6 +55,11 @@ function updateUTM(url: string): string {
         </div>
         <div class="media" relative overflow-hidden max-w-120px min-w-120px>
           <img :src="item.imageUrl" rounded-lg overflow-hidden w-full ha>
+          <div my-y-4px flex flex-justify-end>
```
Comment (jpezninjo) [inline components/timeline/TimelineDiscover.vue:58]: is `my-y-4px` a mistype? I can't see it when I inspect in my dev tools

File: server/api/[server]/recommendations.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ export default defineEventHandler(async (event) => {
     const { locale } = getQuery(event)
 
     const recommendations = await fetch(
-      `https://firefox-api-proxy.readitlater.com/desktop/v1/recommendations?consumer_key=moso-web-dev&locale=${locale}`,
+      `https://mozilla.social/content-feed/moso/v1/discover?locale=${locale}`,
```
Comment (jpezninjo) [inline server/api/[server]/recommendations.ts:6]: 🎉🎉🎉

File: components/timeline/TimelineDiscover.vue (hunk 1 above)
Comment (wtfluckey) [reply components/timeline/TimelineDiscover.vue:22]: I miss explicit imports soooo bad

File: components/timeline/TimelineDiscover.vue (hunk 2 above)
Comment (anthony-liddle) [reply components/timeline/TimelineDiscover.vue:58]: ah yes, thanks, it should be `m-y-4px`. Updating now.

Comment (wtfluckey) [reply components/timeline/TimelineDiscover.vue]: good catch Joseph!

**Review by jpezninjo — APPROVED** [review]

**Review by wtfluckey — APPROVED** [review]

**Pull request merged**